### DISCUSSION TITLE
issue: form is not getting cleared on unmount

### DISCUSSION THREAD

**Discussion by ryub3n** [react-hook-form/react-hook-form#10439] [Q&A] [answered]
### Version Number

7.43.9

### Codesandbox/Expo snack

https://codesandbox.io/s/frosty-cherry-rc6wbl?file=/src/TicketForm.tsx

### Steps to reproduce

Using the sandbox:

1. Select any type to see the form.
2. Enter some values in one of the fields but do not click submit.
3. Now change the type to another.
4. Notice the field value still remains.

### Expected behaviour

The values should be cleared when the type is changed,

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
The inputs itself never really get unmounted, you can call unregister method manually to achieve what you want.

**Comment by ryub3n** [comment]
Can you show me a sample? I want the inputs to be cleared when I change the type.

**Comment by bluebill1049** [comment]
attach `onChange` to the select on each selection change.

```
onChange={() => {
 unregister('...')
}}
```

**Comment by ryub3n** [comment]
For this I'd have to pass `unregister` from the child to the parent component?

**Comment by NishilSB** [comment, marked as answer]
you can use useFormContext [https://www.react-hook-form.com/api/useformcontext/](url) to do this

**Reply by ryub3n** [reply]
This worked, had to tweak it.
